shumai/module/transformer.ts

```diff
@@ -451,3 +460,178 @@ export class TransformerEncoder extends Module {
     return output
   }
 }
+
+/**
+ * A layer of the Transformer decoder, as described by Vaswani et al, consisting of a masked {@link TransformerMultiheadAttention | multi-head} self-attention layer, an unmasked {@link TransformerMultiheadAttention | multi-head} cross-attention layer and a fully-connected feed forward network. All of these use residual connections and are normalised with {@link LayerNorm}.
+ */
+export class TransformerDecoderLayer extends Module {
+  private dim: number
+  private heads: number
+  private attentionDim: number
+  private feedForwardDim: number
+  private maskedMhsa: TransformerMultiheadAttention
```

Comment:
nit (optional): maybe `maskedSelfAttention`? kinda hard to see that `s` hiding in there :P